python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -333,9 +334,25 @@ def updateBlendRecords(
             blend=blend,
             useFlux=useFlux,
         )
-        sourceRecord.setFootprint(heavy)
 
         if updateFluxColumns:
+            if heavy.getArea() == 0:
+                # The source has no flux after being weighted with the PSF
+                # in this particular band (it might have flux in others).
+                sourceRecord.set("deblend_zeroFlux", True)
+                # Create a Footprint with a single pixel, set to zero,
+                # to avoid breakage in measurement algorithms.
+                dtype = observation.images.dtype
+                center = Point2I(heavy.peaks[0]["i_x"], heavy.peaks[0]["i_y"])
+                spanList = [Span(center.y, center.x, center.x)]
+                footprint = afwFootprint(SpanSet(spanList))
+                footprint.setPeakCatalog(heavy.peaks)
+                zeroImage = afwImage(array=np.array([[0]], dtype=np.float32), xy0=center, dtype=dtype)
+                heavy = makeHeavyFootprint(footprint, MaskedImage(zeroImage, dtype=dtype))
```

Comment:
```suggestion
                heavy = HeavyFootprintF(footprint)
                heavy.getImageArray()[0] = 0.0
```
is probably a bit more efficient (no need for `Image` or `MaskedImage` temporaries), but I'm not sure it's enough better to merit another Jenkins run.  Just another way to do it.

Reply:
Good idea. I have to rebase the other packages anyway, so I'll make this change, rebase all of them, and just Jenkins without the ci packages.